index.js

```diff
@@ -285,6 +285,28 @@ function getForceKillAfterTimeout({forceKillAfterTimeout = true}) {
 	return forceKillAfterTimeout;
 }
 
+function handleSpawned(spawned, context) {
+	return new Promise((resolve, reject) => {
+		spawned.on('exit', (code, signal) => {
+			if (context.timedOut) {
+				return reject(Object.assign(new Error('Timed out'), {code, signal}));
```

Comment:
```suggestion
				reject(Object.assign(new Error('Timed out'), {code, signal}));
				return;
```

to make the intention clear